### PR TITLE
Add unittest for Refactor kubelet convertStatusToAPIStatus

### LINKED ISSUE #80875: Refactor kubelet convertStatusToAPIStatus
The kubelet's `convertStatusToAPIStatus()` in https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go iterates `podStatus` 3 times in order to construct the 3 container status lists in `PodStatus`. This could be refactored to make a single pass through `podStatus`.

/kind cleanup
/sig node

### PR DESCRIPTION
**Pull request by Rand01ph**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Add unittest for Refactor kubelet `convertStatusToAPIStatus`
**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
None
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Review by mattjmcnaughton — COMMENTED** [review]
I'll leave the review of the actual test logic to @verb .

But, you can fix at least two out of the three test failures by running `hack/update-gofmt.sh` and `hack/update-bazel.sh` :)

**Comment by matthyx** [comment]
Thanks @Rand01ph !
/assign @verb for review
/assign @tallclair for approval

File: pkg/kubelet/kubelet_pods_test.go [hunk 1]
```diff
@@ -2358,3 +2359,180 @@ func TestTruncatePodHostname(t *testing.T) {
...
+		ID:        pod.UID,
+		Name:      pod.Name,
+		Namespace: pod.Namespace,
+		//		ContainerStatuses: cStatuses,
+	}
+	for _, test := range []struct {
+		name          string
+		podSpec       v1.PodSpec
+		cStatuses     []*kubecontainer.ContainerStatus
+		wantContainer *v1.PodStatus
```
Comment (verb) [inline pkg/kubelet/kubelet_pods_test.go]: Should this be named `wantPodStatus`?

File: pkg/kubelet/kubelet_pods_test.go [hunk 2]
```diff
@@ -2358,3 +2359,180 @@ func TestTruncatePodHostname(t *testing.T) {
...
+								Message: "",
+							},
+						},
+					},
+				},
+			},
+		},
+	} {
+		pod.Spec = test.podSpec
+		podStatus.ContainerStatuses = test.cStatuses
```
Comment (verb) [inline pkg/kubelet/kubelet_pods_test.go]: Create a new `pod` and `podStatus` for each test case rather than reusing them here.

**Review by verb — COMMENTED** [review]
Couple of small nits, but overall LGTM! (I'll add my /lgtm after you push an update)

You probably want to remove the "Fixes #80875" in the PR description so that the issue isn't closed when this merges.

**Comment by Rand01ph** [comment]
> Couple of small nits, but overall LGTM! (I'll add my /lgtm after you push an update)
> 
> You probably want to remove the "Fixes #80875" in the PR description so that the issue isn't closed when this merges.

Thanks, I have remove the "Fixes #80875" in the PR description

**Comment by mattjmcnaughton** [comment]
cc @verb 
/assign @tallclair

**Pull request closed without merging**